hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordScanner.java

```diff
@@ -617,6 +617,8 @@ && compareTimestamps(logBlock.getLogBlockHeader().get(INSTANT_TIME), GREATER_THA
   /**
    * Iterate over the GenericRecord in the block, read the hoodie key and partition path and call subclass processors to
    * handle it.
+   * TODO:
+   * 1. what is the purpose of this method? should the HoodieRecord be added to a queue and consumed by an iterator?
    */
   private void processDataBlock(HoodieDataBlock dataBlock, Option<KeySpec> keySpecOpt) throws Exception {
```

Comment:
This should be the main entry for handling the log blocks, each kind of log block would trigger invocation of specific row-level handling methods.